src/condition_codes.rs

```diff
@@ -1,8 +1,30 @@
 #[derive(Debug, Default)]
 pub struct ConditionCodes {
-    pub z: bool,
-    pub s: bool,
-    pub p: bool,
-    pub cy: bool,
-    pub ac: bool,
+    pub zero: bool,
+    pub sign: bool,
+    pub parity: bool,
+    pub carry: bool,
+    pub accumulator: bool,
+}
+
+impl ConditionCodes {
+    pub fn set_carry(&mut self, carry: bool) {
+        self.carry = carry
+    }
+
+    pub fn reset_carry(&mut self) {
+        self.carry = false
+    }
+
+    pub fn set_sign(&mut self, val: u8) {
+        self.sign = (val & 0x80) == 0x80
+    }
+
+    pub fn set_zero(&mut self, val: u8) {
+        self.zero = (val & 0xFF) == 0
+    }
+
+    pub fn set_parity(&mut self, val: u8) {
+        self.parity = val.count_ones() % 2 == 0
```

Comment:
These will likely conflict with @arnolth so we will need to figure out what to do

Reply:
This is almost exactly what I have but I nested them all under one function as update flags. We can definitely do it this way though. It makes the aux carry and the carry for subtraction a bit simpler.